File: fortran90/day8_part1_2019.f90
program solution
    implicit none
    character(len=1500) :: imageData
    integer :: width, height, layerSize, minZeros, result
    integer :: i, layerStart, zeroCount, oneCount, twoCount
    character :: pixel

    open(10, file='input.txt', status='old')
    read(10, '(A)') imageData
    close(10)

    width = 25
    height = 6
    layerSize = width * height
    minZeros = layerSize + 1
    result = 0

    do i = 1, len_trim(imageData), layerSize
        layerStart = i
        zeroCount = 0
        oneCount = 0
        twoCount = 0

        do while (layerStart <= min(i+layerSize-1, len_trim(imageData)))
            read(imageData(layerStart:layerStart), '(A)') pixel
            layerStart = layerStart + 1

            select case (pixel)
            case ('0')
                zeroCount = zeroCount + 1
            case ('1')
                oneCount = oneCount + 1
            case ('2')
                twoCount = twoCount + 1
            end select
        end do

        if (zeroCount < minZeros) then
            minZeros = zeroCount
            result = oneCount * twoCount
        end if
    end do

    write(*, '(I0)') result
end program solution

File: objectivec/day8_part1_2019.m
#import <Foundation/Foundation.h>

int main() {
    NSError *error;
    NSString *imageData = [NSString stringWithContentsOfFile:@"input.txt" encoding:NSUTF8StringEncoding error:&error];
    
    // Check if file exists or there was an error reading it
    if (!imageData) {
        if (error) {
            NSLog(@"Error reading file: %@", error);
        } else {
            NSLog(@"File 'input.txt' not found or couldn't be read.");
        }
        return 1;
    }

    imageData = [imageData stringByTrimmingCharactersInSet:NSCharacterSet.whitespaceCharacterSet];

    // Check if image data is not empty
    if (imageData.length == 0) {
        NSLog(@"Input data is empty.");
        return 1;
    }

    int width = 25, height = 6;
    int layerSize = width * height;

    int minZeros = layerSize + 1;
    int result = 0;

    // Check if image data length is a multiple of layer size
    if (imageData.length % layerSize != 0) {
        NSLog(@"Invalid input data length.");
        return 1;
    }

    for (int i = 0; i < imageData.length; i += layerSize) {
        NSRange layerRange = NSMakeRange(i, MIN(layerSize, (int)imageData.length - i));
        NSString *layer = [imageData substringWithRange:layerRange];
        int zeroCount = 0, oneCount = 0, twoCount = 0;

        for (int j = 0; j < layer.length; j++) {
            switch ([layer characterAtIndex:j]) {
                case '0':
                    zeroCount++;
                    break;
                case '1':
                    oneCount++;
                    break;
                case '2':
                    twoCount++;
                    break;
                default:
                    NSLog(@"Invalid character in input data: %c", [layer characterAtIndex:j]);
                    return 1;
            }
        }

        if (zeroCount < minZeros) {
            minZeros = zeroCount;
            result = oneCount * twoCount;
        }
    }

    NSLog(@"%d", result);

    return 0;
}